lib/sycamore/sycamore/transforms/text_extraction/pdf_miner.py

```diff
@@ -0,0 +1,108 @@
+from sycamore.data import Element, BoundingBox
+from sycamore.utils.cache import DiskCache
+from io import IOBase
+from typing import BinaryIO, Tuple, List, Union, cast, Any
+from pathlib import Path
+from sycamore.utils.import_utils import requires_modules
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+pdf_miner_cache = DiskCache(str(Path.home() / ".sycamore/PDFMinerCache"))
+
+
+class PDFMinerExtractor:
+    @requires_modules(["pdfminer", "pdfminer.utils"], extra="local-inference")
+    def __init__(self):
+        from pdfminer.converter import PDFPageAggregator
+        from pdfminer.layout import LAParams
+        from pdfminer.pdfinterp import PDFPageInterpreter, PDFResourceManager
+
+        rm = PDFResourceManager()
+        param = LAParams()
+        self.device = PDFPageAggregator(rm, laparams=param)
+        self.interpreter = PDFPageInterpreter(rm, self.device)
+
+    def _open_pdfminer_pages_generator(self, fp: BinaryIO):
+        from pdfminer.pdfpage import PDFPage
+
+        pages = PDFPage.get_pages(fp)
+        for page in pages:
+            self.interpreter.process_page(page)
+            page_layout = self.device.get_result()
+            yield page, page_layout
+
+    @staticmethod
+    def _convert_bbox_coordinates(
+        rect: Tuple[float, float, float, float],
+        height: float,
+    ) -> Tuple[float, float, float, float]:
+        """
+        pdf coordinates are different, bottom left is origin, also two diagonal points defining a rectangle is
+        (bottom left, upper right), for details, refer
+        https://www.leadtools.com/help/leadtools/v19/dh/to/pdf-topics-pdfcoordinatesystem.html
+        """
+        x1, y2, x2, y1 = rect
+        y1 = height - y1
+        y2 = height - y2
+        return x1, y1, x2, y2
+
+    def extract_document(
```

Comment:
Is the idea that this is legacy and will get removed when we don't have a dependency from the managed service? Otherwise, it seems like there is quite a bit of duplication. You could probably implement this as a loop over pdf_to_pages.

Reply:
Yep, exactly. Kept it here and will remove once the managed service has moved.